keras_cv/layers/object_detection/non_max_suppression.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+from keras_cv import bounding_box
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class NonMaxSuppression(tf.keras.layers.Layer):
```

Comment:
If we're creating a brand new NonMaxSupression layer, shouldn't the NmsPredictionDecoder be updated to use this internally?